### DISCUSSION TITLE
2005/11/13/%E7%94%9F%E7%89%A9%E8%BF%9B%E5%8C%96%E6%80%9D%E6%83%B3%E4%B8%8E%E7%A8%8B%E5%BA%8F%E8%AE%BE%E8%AE%A1%E6%80%9D%E6%83%B3%E7%9A%84%E6%AF%94%E8%BE%83

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#55] [Announcements]
# 2005/11/13/%E7%94%9F%E7%89%A9%E8%BF%9B%E5%8C%96%E6%80%9D%E6%83%B3%E4%B8%8E%E7%A8%8B%E5%BA%8F%E8%AE%BE%E8%AE%A1%E6%80%9D%E6%83%B3%E7%9A%84%E6%AF%94%E8%BE%83

阮奇桢

https://blog.qizhen.xyz/2005/11/13/%E7%94%9F%E7%89%A9%E8%BF%9B%E5%8C%96%E6%80%9D%E6%83%B3%E4%B8%8E%E7%A8%8B%E5%BA%8F%E8%AE%BE%E8%AE%A1%E6%80%9D%E6%83%B3%E7%9A%84%E6%AF%94%E8%BE%83

**Comment by ruanqizhen** [comment]
之前博客下的留言：

55380855
2012年05月24日 16:09 编辑
真是太巧了
最初我也是在下棋中想到了可以基于进化论设计出现程序来下棋，当时是大二宿舍好多人在下跳棋，我就想，其实下棋完全不用思考就能够完成，即设计出可以自动进化下棋的程序。
后来偶尔在图书馆才看到，有种通常叫遗传算法的就是干这的。
但对编程知之甚少，也就从来没实践过。不过想想，这应该是个有趣的事。

还有说到病毒，有没有基于进化算法设计的病毒？

但对于博主关于进化论以及转基因的怀疑表示无法苟同。

回复
ruanqizhen
2012年05月25日 08:57 编辑
计算机病毒目前还都是人工编写的，有一些辅助工具帮助制作病毒，但是据我所知没有基于进化算法的病毒。

回复
55380855
2012年05月25日 09:37 编辑
多谢博主介绍，对这个问题我很久很久以前以前想过，不过后来一直没再想起来。今天你倒是解了我的一个疑惑。

回复
mu
2010年03月24日 16:42 编辑
推荐你一本书，约翰。霍兰 的《隐秩序》，遗传算法和人工智能都讲的很明白了。我不知道你的遗传算法是自己想出来的，还是借鉴传统的遗传算法。有兴趣加我MSN聊吧。

回复
帆
2009年02月25日 15:37 编辑
很有启发。难怪看你的文章比较亲切。我也是对生物比较感兴趣，高中凭借生物竞赛进的现在这个名牌大学。学的却不是生物专业。呵呵。

回复
Qizhen
2008年08月4日 10:36 编辑
这是我好几年前的一点想法，之后再没有深入研究。希望它对你有帮助。

回复
jian
2008年08月4日 09:46 编辑
johnnylj98@gmail.com

回复
jian
2008年08月4日 09:45 编辑
我觉得程序段对于大型系统的位置，就像细胞对于生物体的位置。让每一段程序都具有进化能力的想法多令人着迷啊！这说不定才是人工智能的基石。
但是作者提出的问题，“有黑白棋变成五子棋”确实是对进化论反观的一个很大挑战。值得思考。
我刚刚开始读研究生，方向是编译，所以有机会了解到程序运行时的工作，也有可能在程序运行时的时候加入一些进化的内容。如果作者还在继续这方面的思考，希望能给小弟一点入门科普：）

回复
Unknown
2007年04月3日 15:10 编辑
我也想过这个问题，不过是病毒方面，让病毒程序具有某种基因特征，让他们在网络上自由结合,不适应的病毒被杀毒软件杀死，产生的新特征的病毒，会适应并发展下去。

回复
rainraincn
2007年03月14日 17:44 编辑
程序进化没拉 hehe

回复
Qizhen
2006年07月14日 17:33 编辑
非常遗憾的是我花了好长时间编写的遗传算法的黑白棋程序后来遗失了。换了几次电脑之后我再也找不到它了 :(
不过这也进一步促成了我把所有想法，文章，代码都公布到网上，以免不慎丢失，或被遗忘。

回复
hui
2005年12月5日 17:14 编辑
很受启发
